### PR TITLE
fix(userspace/falco): applies FALCO_INSTALL_CONF_FILE as the default …

### LINKED ISSUE #1897: Unnecessary default value for `-c`
**Describe the bug**

The command-line option `-c` has two default values:

```shell
-c                            Configuration file (default /source/falco/falco.yaml, /etc/falco/falco.yaml)
```

Although it's not really a bug, I think `/source/falco/falco.yaml` is misleading. Furthermore, usually users do not expect that `/source/falco/falco.yaml` takes precedence over `/etc/falco/falco.yaml`

Thus, I would like to do not use `/source/falco/falco.yaml` as a default value.

**How to reproduce it**

Store a configuration file at `/source/falco/falco.yaml`, then run `falco` (without `-c` option).

**Expected behaviour**

If `-c` is not specified, `falco` should look for `/etc/falco/falco.yaml` only.

**Environment**

- Falco version: 0.31.0
- System info:

- Cloud provider or hardware configuration:
- OS:

- Kernel:

- Installation method:

**Additional context**

### PR DESCRIPTION
**Pull request by andreabonanno**
…config.

Signed-off-by: Andrea Bonanno <andrea@bonanno.cloud>

**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area engine

> /area rules

> /area tests

> /area proposals

**What this PR does / why we need it**:

Makes Falco defaults to the config file in` /etc/falco/falco.yaml` when the `-c` option is not used, instead of evaluating `/source/falco/falco.yaml` as the config source with highest priority. 

**Which issue(s) this PR fixes**:

Fixes #1897 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by leogr — COMMENTED** [review]
Thank you!

Your solution works, but I'd rather keep the old behavior when `-DCMAKE_BUILD_TYPE=Debug` and use **only** the installation path when `-DCMAKE_BUILD_TYPE=Release`.

Wdyt? :)

**Review by leogr — APPROVED** [review]
LGTM

Thank you!

/approve

File: userspace/falco/falco.cpp [hunk 1]
```diff
@@ -873,21 +877,28 @@ int falco_init(int argc, char **argv)
 		}
 		else
 		{
-			conf_stream.open(FALCO_SOURCE_CONF_FILE);
+
+#ifdef BUILD_TYPE_RELEASE
```
Comment (jasondellaluce) [inline userspace/falco/falco.cpp:881]: From my understanding:
- If `BUILD_TYPE_RELEASE` is defined, we first try to load from `FALCO_INSTALL_CONF_FILE`, and then from `FALCO_SOURCE_CONF_FILE`
- If `BUILD_TYPE_RELEASE` is not defined, we only try to load from `FALCO_SOURCE_CONF_FILE`

Which is not the semantic we seem to describe in the `-help` section. Am I missing something?

Comment (andreabonanno) [reply userspace/falco/falco.cpp]: No worries!

Comment (leogr) [reply userspace/falco/falco.cpp]: Thank you :pray:

**Review by jasondellaluce — APPROVED** [review]

**Pull request merged**